scala-thoughts.md

```diff
@@ -0,0 +1,16 @@
+It's extremely easy to write functional programs in Scala. I didn't ever feel
+the need to reach for imperative programming for any of these problems.
+
+I had a really hard time messing with pairs, though that may be my own
+inexperience.
+
+I really like the way lambdas look as well as the way you can call functions
+without dots and parentheses. They are small changes, but they really make the
+code look like it flows more cleanly when we are chaining many operations. I
+think this was a design choice when functional programming was made to be the
+main paradigm as that is where it cleans the code the most.
+
+I didn't find myself disliking anything in particular about the language, though
+I didn't enjoy constantly looking through the documentation to try and see if
+there was a function to do this thing I was thinking of. Though that will happen
+with learning any language.
```

Comment:
I also had this kind of feeling as well. Also, I agree that it is very natural to write functional programs in Scala.